### ISSUE TITLE
Package request: CVXPY + Clarabel

### ISSUE THREAD

**Issue by phschiele** [pyodide/pyodide#4578]
## 🐍 Package Request

- Package Name and Version: cvxpy/latest, clarabel/latest
- Package URL: https://github.com/cvxpy/cvxpy, https://github.com/oxfordcontrol/Clarabel.rs
- Package Dependencies that needs to be resolved first: All dependencies already available (NumPy, SciPy)

The CVXPY team would love to bring convex optimization to Pyodide (a long standing [user request](https://github.com/pyodide/pyodide/issues/3083))!
This would require bringing over the modeling language (CVXPY) as well as at least one solver (Clarabel).
Below is a short description of what would be required for both.

CVXPY:
- We've recently made it possible to run CVXPY without the custom C++ backend (https://github.com/cvxpy/cvxpy/pull/2363)
- Building pure Python wheels still requires a few lines of changes, which we would maintain in a diff here.

Clarabel:
- Clarabel is written in Rust with a PyO3 interface. The below Dockerfile was used to create a working test wheel
- For one problem type Clarabel relies on BLAS, and the currently generated wheel does not seem to work yet, so this would be the one open issue at the moment

We'd be happy to create a PR to get this started!

```dockerfile
FROM python:3.11.3
ENV EMSDK=/opt/emsdk EMSDK_NODE=/opt/emsdk/node/16.20.0_64bit/bin/node \
    PATH=/opt/emsdk:/opt/emsdk/upstream/emscripten:/opt/emsdk/node/16.20.0_64bit/bin:$PATH
RUN git clone https://github.com/emscripten-core/emsdk.git $EMSDK &&\
    emsdk install 3.1.45 && emsdk activate 3.1.45

# Install rust
RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly -y 

ENV PATH="/root/.cargo/bin:${PATH}"
RUN rustup target add wasm32-unknown-emscripten

# Install pipx
RUN python -m pip install pipx

# Install maturin
RUN pipx install maturin

# Add maturin to PATH
ENV PATH="/root/.local/bin:${PATH}"

# Build Clarabel.rs for wasm
RUN git clone https://github.com/oxfordcontrol/Clarabel.rs &&\
    cd Clarabel.rs &&\
    maturin build --release --target wasm32-unknown-emscripten
```

**Comment by hoodmane** [comment]
> We'd be happy to create a PR to get this started!

PR welcome, thanks @phschiele.

**Comment by phschiele** [comment]
Closed via #4587.
